### DISCUSSION TITLE
(duplicate) FastApi client IP address returns none

### DISCUSSION THREAD

**Discussion by Qwert512** [deta/cloud-docs#750] [Help]
For my fastapi application beeing hosted on deta, i want to get the users IP address to implement a feature to avoid the api beeing spammed. After some quick research i found ip_addr = request.client.host, but when hosting on deta it doesnt work.
Code snippet:

@app.get("/api/test")
async def api_check_uuid(request:Request):
    client_host = request.client
    print(client_host)
    return{"hello":"world"}

returns:
Address(host=None, port=0)
  
When hosting local with uvicorn:
Address(host='127.0.0.1', port=51654)

Does anyone have ideas?
Qwert512

**Comment by lemonyte** [comment]
Duplicate of #282.

**Reply by Qwert512** [reply]
Thanks!

**Comment by lihugang** [comment]
I gave a solution here <https://github.com/deta/cloud-docs/discussions/282#discussioncomment-5017418>
